Stage

evereq opened the pull request:
## Description

Please include a summary of the changes and the related issues.

## Type of Change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Breaking change
-   [ ] Documentation update

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented on my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings

## Previous screenshots

Please add here videos or images of the previous status

## Current screenshots

Please add here videos or images of the current (new) status

joel-kalema committed:
Fix/auth workspace designs (#72)

* fixed the auth pages design

* updated the Sidebar design

* fixed the Side bar Menu Item

* fixed the Side bar Works Item design

* refractor(extension): updated the Sidebar design

* fix(extensions): updated the Side bar Workspaces Item design

* fix: Resolve style issues by following the best practice

* fix (auth): Fixed checkbox onChange handler for proper state management

evereq committed:
Update docker-build-publish-api-dev.yaml (#73)

* Update docker-build-publish-api-dev.yaml

* Update docker-build-publish-api.yaml

joel-kalema committed:
fix(portal): Implement remember me email feature, and error handling

evereq committed:
Merge pull request #74 from ever-co/fix/rememberMe-action

Implement remember me email feature

ani-ali committed:
[UI]: Header updated of integrations and workspce. (#79)

* ui: updated the header of integrations and workspace

* Update apps/portal/components/shared/SCHeader/SCHeader.tsx

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

---------

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ani-ali committed:
[Translation] - Extension: Multi-Language Support. (#87)

* translations of panel is done

* translations of all files is done

* translations of extension popup is done

* translations is completed

* translation of extension keys values are updated with grammer fix

* [Translation] - Web: Multi-Language Support. (#77)

* feat: languages working

* working on translations

* feat: added the whole translations in this app

* feat: added the whole translations in this app

* feat: working on language translations

* first commit

* first commit

* translations finished

* Update apps/portal/i18n/constants.ts

Added Types.

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* code-rabbit: issues fixed

* code-rabbit: issues fixed

* code-rabbit: issues fixed

* code-rabbit: issues fixed in all languages jsons

* code-rabbit: issues fixed in all languages jsons

* code-rabbit: issues fixed in all languages jsons

* code-rabbit: issues fixed in all languages jsons

* code-rabbit: issues fixed in all languages jsons

* code-rabbit: issues fixed in all languages jsons

* Update SortingDropDown.tsx

* Update GetScreenshotUrlModal.tsx

updated the remaining text.

* Update useGenerateShareLink.ts

* Update useGenerateShareLink.ts

* translations of web is completed

* translation of extension is done.

* translation of web is done. with new texts

* translation of web is done. with new texts

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Co-authored-by: Ruslan Konviser <evereq@gmail.com>

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Co-authored-by: Ruslan Konviser <evereq@gmail.com>

dependabot[bot] committed:
Bump @nestjs/common from 10.4.4 to 10.4.16

Bumps [@nestjs/common](https://github.com/nestjs/nest/tree/HEAD/packages/common) from 10.4.4 to 10.4.16.
- [Release notes](https://github.com/nestjs/nest/releases)
- [Commits](https://github.com/nestjs/nest/commits/v10.4.16/packages/common)

---
updated-dependencies:
- dependency-name: "@nestjs/common"
  dependency-version: 10.4.16
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump webpack-dev-server from 5.1.0 to 5.2.1

Bumps [webpack-dev-server](https://github.com/webpack/webpack-dev-server) from 5.1.0 to 5.2.1.
- [Release notes](https://github.com/webpack/webpack-dev-server/releases)
- [Changelog](https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-server/compare/v5.1.0...v5.2.1)

---
updated-dependencies:
- dependency-name: webpack-dev-server
  dependency-version: 5.2.1
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump next from 14.2.21 to 14.2.30

Bumps [next](https://github.com/vercel/next.js) from 14.2.21 to 14.2.30.
- [Release notes](https://github.com/vercel/next.js/releases)
- [Changelog](https://github.com/vercel/next.js/blob/canary/release.js)
- [Commits](https://github.com/vercel/next.js/compare/v14.2.21...v14.2.30)

---
updated-dependencies:
- dependency-name: next
  dependency-version: 14.2.30
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

evereq committed:
Bump next from 14.2.21 to 14.2.30 (#88)

Bumps [next](https://github.com/vercel/next.js) from 14.2.21 to 14.2.30.
- [Release notes](https://github.com/vercel/next.js/releases)
- [Changelog](https://github.com/vercel/next.js/blob/canary/release.js)
- [Commits](https://github.com/vercel/next.js/compare/v14.2.21...v14.2.30)

---
updated-dependencies:
- dependency-name: next
  dependency-version: 14.2.30
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

evereq committed:
Merge pull request #89 from ever-co/dependabot/npm_and_yarn/nestjs/common-10.4.16

Bump @nestjs/common from 10.4.4 to 10.4.16

ani-ali committed:
Merge pull request #67 from ever-co/66-fixui-change-ui-of-started-folders

[UI] - Web: Convert the UI of Favorite Page

ani-ali committed:
Merge pull request #81 from ever-co/80-ui-extension-updated-header-of-integrations-and-workspaces

 [UI] - Extension: Updated header of integrations and workspaces.

ani-ali committed:
[Build] - Web: Error on language import path. (#91)

* build: issue fix of languages import path.

* build: issue fix of languages import path.

evereq committed:
Bump webpack-dev-server from 5.1.0 to 5.2.1 (#86)

Bumps [webpack-dev-server](https://github.com/webpack/webpack-dev-server) from 5.1.0 to 5.2.1.
- [Release notes](https://github.com/webpack/webpack-dev-server/releases)
- [Changelog](https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-server/compare/v5.1.0...v5.2.1)

---
updated-dependencies:
- dependency-name: webpack-dev-server
  dependency-version: 5.2.1
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ani-ali committed:
[UI] - Extension: Convert the UI of Favorite Page. (#84)

* added favorites in extension

* Update MyImages.tsx

* added favorites in extension

* code-rabbit: all the extension issues fix for favorite folders

* code-rabbit: all the extension issues fix for favorite folders

* feat: issue of translation in favorites is fixed

* feat: issue of translation in favorites is fixed

* feat: issue of translation in favorites is fixed

dependabot[bot] committed:
Bump jspdf from 2.5.1 to 3.0.1 (#75)

Bumps [jspdf](https://github.com/MrRio/jsPDF) from 2.5.1 to 3.0.1.
- [Release notes](https://github.com/MrRio/jsPDF/releases)
- [Changelog](https://github.com/parallax/jsPDF/blob/master/RELEASE.md)
- [Commits](https://github.com/MrRio/jsPDF/compare/v2.5.1...v3.0.1)

---
updated-dependencies:
- dependency-name: jspdf
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Anish <105014311+Anishali2@users.noreply.github.com>

ani-ali committed:
feat: latest pnpm.lock file is updated

evereq committed:
Merge pull request #93 from ever-co/92-build-pnpm-lock-file-update

[Build] : PNPM lock file update.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ever-co/ever-rec?pullRequest=94) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/ever-co/ever-rec?pullRequest=94) before we can accept your contribution.<br/>**5** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: joel-kalema<br/>:white_check_mark: paradoxe35<br/>:white_check_mark: adkif<br/>:white_check_mark: evereq<br/>:white_check_mark: ani-ali<br/>:x: dependabot[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ever-co/ever-rec?pullRequest=94) it.</sub>

evereq committed:
Update README.md

evereq committed:
Update CONTRIBUTING.MD

evereq committed:
chore: add Desktop App repo as submodule

dependabot[bot] committed:
Bump multer from 1.4.5-lts.1 to 2.0.1 (#85)

* Bump multer from 1.4.5-lts.1 to 2.0.1

Bumps [multer](https://github.com/expressjs/multer) from 1.4.5-lts.1 to 2.0.1.
- [Release notes](https://github.com/expressjs/multer/releases)
- [Changelog](https://github.com/expressjs/multer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/expressjs/multer/compare/v1.4.5-lts.1...v2.0.1)

---
updated-dependencies:
- dependency-name: multer
  dependency-version: 2.0.1
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update pnpm-lock.yaml

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Anish <105014311+Anishali2@users.noreply.github.com>

ani-ali committed:
[Build] - Web: Error on language import path again. (#96)

* feat: path of languages are updated again

* feat: path of languages are updated again

joel-kalema commented:
I did it sir @evereq

ani-ali committed:
[Build] : Portal Build issue exist. (#98)

* feat: build issue fixed in 3 files

* feat: build issue fixed in 3 files

* feat: build issue fixed in 3 files

* feat: build issue fixed in 3 files

* feat: build issue fixed in 3 files

* feat: issue of sidebar is fixed

* feat: issue of sidebar is fixed

dependabot[bot] committed:
Bump form-data from 4.0.1 to 4.0.4

Bumps [form-data](https://github.com/form-data/form-data) from 4.0.1 to 4.0.4.
- [Release notes](https://github.com/form-data/form-data/releases)
- [Changelog](https://github.com/form-data/form-data/blob/master/CHANGELOG.md)
- [Commits](https://github.com/form-data/form-data/compare/v4.0.1...v4.0.4)

---
updated-dependencies:
- dependency-name: form-data
  dependency-version: 4.0.4
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

evereq committed:
Bump form-data from 4.0.1 to 4.0.4 (#99)

Bumps [form-data](https://github.com/form-data/form-data) from 4.0.1 to 4.0.4.
- [Release notes](https://github.com/form-data/form-data/releases)
- [Changelog](https://github.com/form-data/form-data/blob/master/CHANGELOG.md)
- [Commits](https://github.com/form-data/form-data/compare/v4.0.1...v4.0.4)

---
updated-dependencies:
- dependency-name: form-data
  dependency-version: 4.0.4
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump multer from 2.0.1 to 2.0.2

Bumps [multer](https://github.com/expressjs/multer) from 2.0.1 to 2.0.2.
- [Release notes](https://github.com/expressjs/multer/releases)
- [Changelog](https://github.com/expressjs/multer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/expressjs/multer/compare/v2.0.1...v2.0.2)

---
updated-dependencies:
- dependency-name: multer
  dependency-version: 2.0.2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

evereq committed:
Merge pull request #101 from ever-co/dependabot/npm_and_yarn/multer-2.0.2

Bump multer from 2.0.1 to 2.0.2

adkif committed:
[Feat] Enhance API (#100)

* feat(auth): enhance authentication and token management

Refactor Firebase Admin SDK initialization into a dedicated module for
improved setup and global provision.
Introduce TokenService to centralize ID token verification and refresh
logic, enhancing security and robustness.
Implement new authentication endpoints for token refreshing, email
verification link generation, and user reauthentication.
Standardize authentication DTOs and update AuthGuard to use TokenService.
Add @nestjs/swagger dependency for API documentation.
Include Codacy rules for AI development workflow guidelines.

* fix(auth): prevent error from missing authorization header

The previous implementation would throw a TypeError if the
'authorization' header was undefined or null. This change
uses optional chaining to safely access the header,
preventing application crashes when the header is absent.

* build(deps): upgrade nestjs to v11 and update core libraries

Updated NestJS framework from v10 to v11, along with various
other core dependencies such as Sentry, Mux, and Node.js client
libraries.
Refactored Sentry initialization to use the official @sentry/nestjs
package and a dedicated global instrument file.
Introduced email validation for verification link generation and
corrected a typo in the controller method name. The delete app event
endpoint was also refactored for clearer IP handling.
Swagger API documentation setup was added.

* docs(auth): document DTOs with Swagger API properties

This change adds `@ApiProperty` and `@ApiPropertyOptional` decorators
to all relevant properties within the authentication DTOs.
This provides detailed descriptions, examples, and validation rules
for the automatically generated OpenAPI (Swagger) documentation,
improving clarity for API consumers.

* fix: correct user deletion flow and module import paths

Reorder user deletion logic in AuthService to ensure user data
is removed from the database before the Firebase Authentication
user is deleted. This prevents potential orphaned data if the
database removal fails.
Rename 'drve.module.ts' to 'drive.module.ts' and update its
import path in 'app.module.ts' to resolve incorrect module
references.
Update 'FIREBASE_ADMIN' constant to use a unique Symbol for
dependency injection. This enhances robustness and prevents
potential token collisions.

* style(mux): reorder imports

Align import order with project conventions.

* fix(auth): improve error handling and validation in authentication flow

Refactor error handling in AuthService to provide specific feedback for various Firebase authentication errors. Update LoginDto to include validation decorators for email and password fields, ensuring better input validation. Correct a typo in the generateEmailVerificationLink method name for clarity. Additionally, streamline the deleteAppEventDefault method in LogController for improved IP handling.

* feat(auth): add email ownership guard

The `EmailOwnershipGuard` prevents authenticated users from attempting to modify
or initiate actions (e.g., email change, password update, verification links)
for an email address not associated with their account. This enhances security
by ensuring the email in the request body matches the authenticated user's email.
Additionally, this commit includes minor code cleanups:
- Remove redundant `await` keywords in controller methods.
- Add `readonly` modifier to DTO properties for immutability.

* feat(auth): add validation to auth dtos

Integrates class-validator decorators to enforce strict input
validation for authentication-related data transfer objects.
This ensures that incoming requests conform to defined schemas,
improving data integrity and API reliability.

* refactor(auth): remove unused signup dto and simplify email update

Removes the `signup.dto.ts` file as it is no longer required for
user registration. The `EmailOwnershipGuard` was also removed from the
`update-email` endpoint, simplifying the guard chain. Necessary
ownership checks are now handled within the `AuthService` logic.

* feat(auth): implement robust auth architecture and extend capabilities

This commit introduces a comprehensive refactor of the authentication
module, breaking down the monolithic AuthService into highly specialized
and cohesive services.
Key changes include:
- Splitting AuthService logic into:
- AuthenticationService (login, register, password change)
- EmailService (email verification, password reset flows)
- GoogleAuthService (Google login processing)
- UserProfileService (user data in database, avatar upload)
- UserService (direct Firebase Auth/Admin SDK interactions)
- Introducing AuthOrchestratorService as a facade for AuthController,
simplifying controller logic and centralizing auth operations.
- Abstracting Firebase Admin SDK and Identity Toolkit REST API calls
into dedicated FirebaseAdminService, FirebaseAuthService, and
FirebaseRestService for better separation of concerns and testability.
- Extending API capabilities by exposing new endpoints for:
- Sending password reset emails
- Generating/sending email verification links
- Verifying emails with codes
- Resending verification emails
- Setting user custom claims
- Enhancing Firebase module configuration, initialization, and type
definitions for improved robustness and maintainability.
- Migrating SharedService to its own dedicated module.

* feat(auth): add email verification status to user response

This commit enhances the authentication flow by retrieving the user's email verification status from Firebase and including it in the response during login and user metadata retrieval. This addition improves user experience by providing immediate feedback on email verification status.

* refactor(auth): remove AuthService and enhance user response with email verification status

This commit deletes the AuthService, consolidating its functionality into specialized services for improved maintainability. Additionally, it updates the user response during login and reauthentication to include the email verification status, enhancing user experience by providing immediate feedback on their verification state.

* fix(user-profile): update user profile retrieval after updates

This commit adds a step to retrieve the updated user profile from the database after applying changes. This ensures that the latest user data is available for subsequent operations, enhancing the accuracy of user information handling.

* refactor(auth): simplify email update process by removing EmailOwnershipGuard

This commit removes the EmailOwnershipGuard from the update email endpoint, streamlining the authentication flow. Additionally, it corrects a typo in the sign-in methods retrieval, ensuring consistency in the codebase. The password update process is enhanced to invalidate previous sessions for better security practices.

* refactor(auth): rename newPassword to password in DTO and controller

This commit updates the `UpdatePasswordDto` and the `AuthController` to rename the `newPassword` field to `password`, ensuring consistency in naming conventions across the authentication module. This change enhances clarity and aligns with the expected input structure for password updates.

* feat(auth): implement Firebase Identity Toolkit for user registration

This commit updates the user registration process to utilize Firebase Identity Toolkit for signing up users, enhancing error handling and ensuring display names are set securely using the Admin SDK. The user data is now structured to include tokens and verification status, improving the overall authentication flow and user experience.

* refactor(auth): update refresh token endpoint and clean up imports

This commit modifies the `refreshToken` method in the `AuthController` to use a GET request instead of POST, enhancing clarity in its purpose. Additionally, it updates the `TokenService` to remove unused dependencies and integrates the `FirebaseRestService` for token handling, improving the overall structure and maintainability of the authentication module. Minor formatting adjustments were also made for consistency.

* feat(auth): add email verification endpoint and refactor email service

This commit introduces a new endpoint in the AuthController for verifying user emails, enhancing the authentication flow. The AuthOrchestratorService is updated to include a verifyEmail method, which interacts with the EmailService to check email verification status. Additionally, the EmailService is refactored to streamline the password reset email process and improve error handling, ensuring a more robust user experience.

* refactor(auth): standardize authentication api routes and params

Rename `newPassword` parameter to `password` in `changeUserPasswordAPI` for
consistency.
Update `changeUserPasswordAPI` endpoint from `/api/v1/auth/update-pass` to
`/api/v1/auth/password` and `changeUserEmailAPI` endpoint from
`/api/v1/auth/update-email` to `/api/v1/auth/email` to standardize API
resource paths.

* refactor(auth): migrate to dedicated firebase auth services

Decouple direct Firebase Admin SDK usage by injecting and utilizing
dedicated Firebase authentication services. This improves testability,
modularity, and adheres to dependency injection principles.
Additionally, remove unnecessary 'await' keywords from event emitter
calls in Google authentication service, as 'emit' is typically a
synchronous or fire-and-forget operation.

* feat(auth): enhance Google login token handling

Refactor Google authentication to use Firebase Identity Toolkit's
signInWithIdp endpoint. This centralizes the token exchange logic
within the FirebaseAuthService and provides the client with standard
Firebase idToken, refreshToken, and expiresAt upon successful login.
- Exchange Google ID token directly with Firebase instead of verifying
it via Firebase Admin SDK.
- Update the login response to include Firebase idToken, refreshToken,
and expiresIn/expiresAt.
- Improve type safety for Google login credentials by changing type
from 'any' to 'string'.
- Make Firebase Identity Toolkit base URL configurable for better
environment management.
- Standardize formatting and error logging across Firebase REST
service methods.

* feat(portal): add OAuth redirect handler for desktop app

This page acts as an intermediary for OAuth authentication flows. It
parses the OAuth tokens from the URL fragment after a successful
redirect from an identity provider.
The extracted tokens are then used to construct a custom deep link
(ever-rec://oauth) which is used to redirect the browser and launch
the Electron desktop application, passing the authentication data
directly to it. This enables seamless login for the desktop app via
a web-based OAuth flow.

* chore(deps): update pnpm dependencies and tidy lockfile

Run pnpm install to update specified dependency versions and optimize the
lockfile structure by removing duplicate entries and outdated deprecated
warnings.

* feat(portal): enhance desktop app authorization deeplink

Implement automatic and manual deep link redirection for desktop
application authorization.
The `authorize` page now attempts to automatically open the desktop
application using a configurable deep link protocol defined by
`NEXT_PUBLIC_DESKTOP_PROTOCOL`. If the automatic redirection fails,
a manual "Open App" button is displayed, allowing users to
initiate the redirect themselves.
This change significantly improves the user experience when logging
into the desktop app via the web portal by providing a robust
redirection mechanism with a clear fallback.

* feat(firebase): make request URI configurable

Replace hardcoded 'http://localhost' values in Firebase Identity Toolkit
API calls with a configurable value. This allows the API to be
configured correctly for different environments (e.g., production
deployments) where 'localhost' is not valid.
The `continueUri` and `requestUri` parameters are now sourced from the
`FIREBASE_REQ_URI` environment variable via the firebase config.

* chore(firebase): remove unused import and reformat config

The `supertest` import was removed as it was an unused testing dependency
found in the Firebase configuration file. Additionally, the
`passwordResetEmail` configuration line was reformatted for improved
readability.

* feat(security): improve deep link validation and auth token expiration

Introduce secure deep link validation in the portal to prevent open redirect vulnerabilities by whitelisting allowed protocols.
Add a new utility to correctly calculate and store absolute expiration dates for authentication tokens. This ensures all `expiresAt` fields in auth responses are precise `Date` objects.
Minor improvements:
* Refactor array method `includes` to `some` for consistency.
* Adjust code indentation for improved readability in multi-line conditions.
* Update button text color on the authorize page for better contrast.

* fix(authorize): harden deep link redirect validation

Restrict allowed protocols for deep link redirection to only the desktop application protocol. Previously, HTTPS and HTTP URLs were also permitted, which was not intended for a flow designed to handle desktop app deep links and could lead to unintended browser redirects.
The `allowedProtocols` check was migrated from an array to a Set for improved lookup performance.
Additionally, add `aria-live="polite"` to the processing status heading for better accessibility with screen readers. Imports were also reordered for consistency.

* fix(deeplink): fail gracefully when desktop protocol is missing

Add a check to ensure the allowed protocols set is not empty.
This prevents the deeplink validation from incorrectly passing or
encountering issues when the NEXT_PUBLIC_DESKTOP_PROTOCOL environment
variable is not configured.

* fix(api/utils): add robust input validation to getExpirationTime

Add comprehensive input validation to the `getExpirationTime` utility.
Previously, invalid `expiresIn` values could lead to unexpected `NaN` results
or incorrect date calculations. This change introduces explicit checks for:
- Blank or whitespace strings.
- Non-finite numbers (NaN, Infinity, -Infinity).
- Negative numbers.
Invalid inputs now throw specific `TypeError` or `RangeError` instances with
descriptive messages, enhancing the function's robustness and clarity.
JSDoc comments were also added to document expected behavior and error types.

* chore: clean up unused firebase methods and portal text

Remove several Firebase REST service methods that are no longer used
by the application, simplifying the API service.
Update the redirect message and button text on the authorize page
to specifically mention "Ever Rec Desktop" for better clarity.

* refactor(firebase): relocate requestUri to identity toolkit config

Move `requestUri` from the top-level firebase config to the `firebase.identityToolkit`
object. This change improves the logical grouping of settings specific to the Firebase
Identity Toolkit API.
Update usages in `FirebaseAuthService` to retrieve `requestUri` from its new nested path
and use `getOrThrow` for stricter configuration access. The default local development
value is also updated from `localhost` to `http://localhost` to ensure it's a valid
absolute URL for callback purposes.

dependabot[bot] committed:
Bump axios from 0.28.1 to 0.30.0

Bumps [axios](https://github.com/axios/axios) from 0.28.1 to 0.30.0.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v0.30.0/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v0.28.1...v0.30.0)

---
updated-dependencies:
- dependency-name: axios
  dependency-version: 0.30.0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump form-data from 2.3.3 to 4.0.4

Bumps [form-data](https://github.com/form-data/form-data) from 2.3.3 to 4.0.4.
- [Release notes](https://github.com/form-data/form-data/releases)
- [Changelog](https://github.com/form-data/form-data/blob/master/CHANGELOG.md)
- [Commits](https://github.com/form-data/form-data/commits/v4.0.4)

---
updated-dependencies:
- dependency-name: form-data
  dependency-version: 4.0.4
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

adkif committed:
chore(deps): update multer and gapi client dependencies

Bump multer to 2.0.2 and @maxim_mazurok/gapi.client.discovery-v1 to
0.2.20200806 in the pnpm lockfile.

evereq committed:
chore(deps): update multer and gapi client dependencies (#104)

Bump multer to 2.0.2 and @maxim_mazurok/gapi.client.discovery-v1 to
0.2.20200806 in the pnpm lockfile.

evereq committed:
Merge pull request #102 from ever-co/dependabot/npm_and_yarn/axios-0.30.0

Bump axios from 0.28.1 to 0.30.0

evereq committed:
Merge pull request #103 from ever-co/dependabot/npm_and_yarn/form-data-4.0.4

Bump form-data from 2.3.3 to 4.0.4

adkif committed:
[Fix]: Portal build  (#105)

* chore(tooling): configure linting, update dependencies and optimize media

Integrate comprehensive ESLint and Prettier configurations to enhance
code quality and consistency. Enable TypeScript-specific rules and refine
`no-unused-vars` to a warning, ignoring prefixed arguments. Remove outdated
`eslint-disable` comments.
Upgrade `konva` and add `canvas` dependencies. Implement `pnpm.overrides`
to ensure consistent `@types/react` versions across the monorepo.
Optimize media component loading by using Next.js dynamic imports with
`ssr: false` for `PlyrPlayer`. This ensures the component is loaded only
on the client-side, reducing initial bundle size and preventing server-side
rendering issues.
Refactor TypeScript imports to use `import type` for type-only declarations.

* chore(deps): make canvas optional and remove pnpm overrides

Move the `canvas` package from direct `dependencies` to `optionalDependencies`
within the `portal` app.
This change improves installation robustness by allowing the application to
function even if `canvas` (which has native dependencies) fails to build
or install on a specific environment.
Additionally, remove the no longer necessary `pnpm.overrides` for
`@types/react` and `@types/react-dom`.
The `pnpm-lock.yaml` has been updated to reflect these changes.

evereq committed:
Update Dockerfile

evereq committed:
Update Dockerfile

evereq committed:
Update docker-build-publish-api-dev.yaml (#106)

* Update docker-build-publish-api-dev.yaml

* Update docker-build-publish-api.yaml

* Create deploy-do-dev.yml

* Create deploy-do-prod.yml

* Create k8s-manifest.dev.yaml

* Create k8s-manifest.prod.yaml

* Delete .deploy/k8s/k8s-manifest.yaml

evereq committed:
Update k8s-manifest.dev.yaml (#107)

* Update k8s-manifest.dev.yaml

* Update k8s-manifest.prod.yaml

* Update k8s-manifest.dev.yaml

* Update k8s-manifest.prod.yaml

* Update k8s-manifest.prod.yaml

* Update k8s-manifest.dev.yaml

* Update k8s-manifest.prod.yaml

* Update .deploy/k8s/k8s-manifest.dev.yaml

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

---------

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

paradoxe35 committed:
remove i18n directory from .dockerignore (#108)
